lambdas/postcode-lookup/src/main/java/uk/gov/di/ipv/cri/address/api/service/PostcodeLookupService.java

```diff
@@ -175,10 +177,18 @@ public List<CanonicalAddress> lookupPostcode(String postcode)
                 new ObjectMapper().readValue(response.body(), postcodeResponse.getClass());
 
         // Map the postcode response to our model
-        return postcodeResponse.getResults().stream()
-                .filter(result -> result.getDpa() != null)
-                .map(result -> result.getDpa().toCanonicalAddress())
-                .collect(Collectors.toList());
+        return Optional.ofNullable(postcodeResponse.getResults())
+                .map(
+                        results ->
+                                results.stream()
+                                        .filter(result -> result != null && result.getDpa() != null)
```

Comment:
Sonar wants a test to cover the `.filter(result -> result != null && result.getDpa() != null)`